File: HackerRank-Data-Structures/Arrays & Linked Lists/Dynamic Array.py
def dynamicArray(n, queries):
    lastAnswer = 0
    array =  []
    for i in range (0, n):
        array.append([])
    res =[]
    for i in queries:
        index = (i[1] ^ lastAnswer) % n
        if i[0] == 1:
            array[index].append(i[2])
        else:
            lastAnswer = array[index][i[2] % len(array[index])]
            res.append(lastAnswer)
    return res
            

File: HackerRank-Data-Structures/Arrays & Linked Lists/Dynamic Array.cs
public static List<int> dynamicArray(int n, List<List<int>> queries)
    {
        int lastAnswer = 0;
        List<List<int>> array =  new List<List<int>>();
        for(int i = 0 ; i < n ; i++){
            array.Add(new List<int>());
        }
        List<int> res = new List<int>();
        foreach(List<int> ele in queries){
            int index = (ele[1] ^ lastAnswer) % n;
            if (ele[0] == 1)
                array[index].Add(ele[2]);
            else
            {
                lastAnswer = array[index][ele[2] % array[index].Count];
                res.Add(lastAnswer);
            }
                
        }
            
        return res;
    }